### PR TITLE
[Mailer] Fix BC breaking event name change

### LINKED ISSUE #47383: `MailerAssertionsTrait::getMailerMessage` returns messages that aren't rendered (BC break)
### Symfony version(s) affected

6.2.0

### Description

Prior to 6.2.0, a call to `self::getMailerMessage(0)` would return the only message that was sent. After https://github.com/symfony/symfony/pull/47191 was merged, this returns a message from the `QueueingMessageEvent`, which means it hasn't been rendered via `MessageListener` yet.

Given this is just in tests, it's not a critical BC break, but it is a BC break. I'm not sure if `getMailerMessage` and `getMailerMessages` is really intended to be returning `RawMessage`s from `QueueingMessageEvent` or not.

### How to reproduce

Create `templates/test_email.html.twig`

```
<?php

use Symfony\Bridge\Twig\Mime\TemplatedEmail;
use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
use Symfony\Component\Mailer\MailerInterface;

class TestCase extends KernelTestCase
{
    public function testItSendsAnEmail(): void
    {
        $mailer = self::getContainer()->get(MailerInterface::class);

        $email = (new TemplatedEmail())
            ->from('from@example.com')
            ->to('user@example.com')
            ->htmlTemplate('test_email.html.twig');

        $mailer->send($email);

        $email = self::getMailerMessage(0);
        self::assertNotNull($email->getHtmlBody());
    }
}
```

### Possible Solution

Change `getMailerMessage` and `getMailerMessages` to filter the events to remove `QueueingMessageEvent`.

### Additional Context

_No response_

### LINKED ISSUE #47989: [Mailer] 6.2 dev - MessageEvent listener not called
### Symfony version(s) affected

6.2-dev

### Description

Working fine on current 6.2 branch.

I define an event listener like this:
```
$services
        ->set(MessageEventListener::class)
        ->tag('kernel.event_listener', ['event' => MessageEvent::class])
        ->args(
            [
                '%env(MAILER_EMAIL)%',
            ]
        );
```
However the handler is no longer being called.

### How to reproduce

I'm trying to work out why the `AbstractTransport::send` method does not appear to be called anymore and why the `MessageEvent` event listener would not be fired.

### Possible Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by chalasr**
| Q             | A
| ------------- | ---
| Branch?       | 6.2
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix #47989, Fix #47383
| License       | MIT
| Doc PR        | -

While I get the motivation for introducing this `QueuingMessageEvent` child event in https://github.com/symfony/symfony/pull/47191, it's a BC break that is difficult to circumvent due to the fact it's an event (the parent may still be used as it's not deprecated). 
So I propose to revert its addition and just add the needed stamp-related methods to the original event instead.

### CODE REVIEW

File: src/Symfony/Component/Mailer/Event/QueuingMessageEvent.php [hunk 1]
```diff
@@ -20,7 +20,7 @@
  *
  * @author Fabien Potencier <fabien@symfony.com>
  */
-final class QueuingMessageEvent extends MessageEvent
+final class MessageEvent extends MessageEvent
```
Comment (ogizanagi) [inline src/Symfony/Component/Mailer/Event/QueuingMessageEvent.php]: I bet you meant to remove this file instead, right?

Comment (chalasr) [reply src/Symfony/Component/Mailer/Event/QueuingMessageEvent.php]: good game :)

**Comment by nesl247** [comment]
@chalasr will this fix #47383 as well like I think it will?

**Comment by chalasr** [comment]
@nesl247 Yes it does, thanks for noticing and testing dev branches.

File: src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php [hunk 2]
```diff
@@ -50,7 +49,7 @@ public static function getSubscribedEvents(): array
     {
         return [
             MessageEvent::class => ['onMessage', -255],
-            QueuingMessageEvent::class => ['onMessage', -255],
+            MessageEvent::class => ['onMessage', -255],
```
Comment (Kocal) [inline src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php]: To be removed

File: src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php [hunk 3]
```diff
@@ -14,7 +14,6 @@
 use Symfony\Component\EventDispatcher\EventSubscriberInterface;
 use Symfony\Component\Mailer\Event\MessageEvent;
 use Symfony\Component\Mailer\Event\MessageEvents;
```
Comment (Kocal) [inline src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php:16]: To be removed

File: src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php (hunk 2 above)
Comment (chalasr) [reply src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php]: Ah yes, nice catch 👍

File: src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php (hunk 3 above)
Comment (chalasr) [reply src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php:16]: Nope! (extra `s` :))

Comment (chalasr) [reply src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php:16]: Also, no threats please 😋
Screenshot 2022-10-25 at 22 44 22

Comment (Kocal) [reply src/Symfony/Component/Mailer/EventListener/MessageLoggerListener.php:16]: Oh, my bad :p

**Review by ogizanagi — APPROVED** [review]

File: src/Symfony/Component/Mailer/Event/MessageEvent.php [hunk 4]
```diff
@@ -64,4 +68,17 @@ public function isQueued(): bool
     {
         return $this->queued;
     }
+
+    public function addStamp(StampInterface $stamp): void
+    {
```
Comment (fabpot) [inline src/Symfony/Component/Mailer/Event/MessageEvent.php:74]: We should probably throw an exception when `queued` is `false` as these stamps won't be used at all in that case.

**Review by fabpot — CHANGES_REQUESTED** [review]

File: src/Symfony/Component/Mailer/Event/MessageEvent.php [hunk 5]
```diff
@@ -64,4 +69,24 @@ public function isQueued(): bool
...
+        if (!$this->queued) {
+            throw new LogicException(sprintf('Cannot call `%s()` on a message that is not meant to be queued. Set $queued to `true` in the constructor.', __METHOD__));
```
Comment (fabpot) [inline src/Symfony/Component/Mailer/Event/MessageEvent.php]: I would remove the second part which does not seem relevant for end user code.

File: src/Symfony/Component/Mailer/Event/MessageEvent.php (hunk 4 above)
Comment (chalasr) [reply src/Symfony/Component/Mailer/Event/MessageEvent.php:74]: done

File: src/Symfony/Component/Mailer/Event/MessageEvent.php (hunk 5 above)
Comment (chalasr) [reply src/Symfony/Component/Mailer/Event/MessageEvent.php]: done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @chalasr.

**Comment by PhilETaylor** [comment]
Sorry to hijack this thread, but wanted to confirm that after this merge, the profiler is still confused and showing 2 emails sent in the left menu and 1 queued and 1 sent. 

Is this on purpose or something else that needs fixing?

My code is only sending one email. 

```
  $message = new Email();
        $message->to('phil@phil-taylor.com')
            ->from('phil@phil-taylor.com')
            ->subject('Mailer Test')
            ->text('This is just a test');

        $mailer->send($message);
```

Screenshot 2022-10-26 at 11 40 27

**Comment by chalasr** [comment]
I'll have a look today, thanks for telling us.

**Comment by chalasr** [comment]
@PhilETaylor I can't reproduce on a basic setup sorry, I get only one queued email reported in the profiler panel. Please make sure it's not an issue in your code and if it's not, consider opening an issue with a minimal application that allows reproducing the issue. Thanks

**Comment by PhilETaylor** [comment]
yes, 30 seconds ago I tested my symfony demo site and I could also not reproduce it - dang it! Sorry for wasting your time, I'll dig deeper. 

The `[main] transport` is confusing though, as my only transport is the `smtp://mailhog...` one... 

edit: Should also say, 6.1 gives only 1 email in the profiler, not 2 - dang it :) :)

**Comment by PhilETaylor** [comment]
ok I fired up the laptop which has never seen 6.2 and now I can replicate my issue in my app only in `6.1.4` - sorry for the noise... only one email is every sent to the `MAILER_DSN` but something is also queuing an email - ill shut up now and go debug more ... sorry for wasting your time on this side show.

**Comment by alexander-schranz** [comment]
I'm running into a similar issue but in tests. Looks like this is an issue with the profiler when mailer and messenger component is installed. But the default messenger is a `sync` bus, what confused me more is that I don't have any message_bus configured it is still `framework.mailer.message_bus: null` configured.

**Pull request merged**